platforms/android/android.toolchain.cmake

```diff
@@ -1198,11 +1207,11 @@ if (ARM64_V8A )
  if( NOT ANDROID_COMPILER_IS_CLANG )
   set( ANDROID_CXX_FLAGS_RELEASE "${ANDROID_CXX_FLAGS_RELEASE} -funswitch-loops -finline-limit=300" )
  endif()
-elseif( ARMEABI OR ARMEABI_V7A)
+elseif( ARMEABI OR ARMEABI_V7A OR ARMEABI_V7A_HARD)
  set( ANDROID_CXX_FLAGS "${ANDROID_CXX_FLAGS} -funwind-tables" )
  if( NOT ANDROID_FORCE_ARM_BUILD AND NOT ARMEABI_V6 )
   set( ANDROID_CXX_FLAGS_RELEASE "-mthumb -fomit-frame-pointer -fno-strict-aliasing" )
-  set( ANDROID_CXX_FLAGS_DEBUG   "-marm -fno-omit-frame-pointer -fno-strict-aliasing" )
+  set( ANDROID_CXX_FLAGS_DEBUG   "-mthumb -fno-omit-frame-pointer -fno-strict-aliasing" )
```

Comment:
I think this line should be reverted. NDK sets arm mode for debug builds (I've checked _r10e_ / _armeabi-v7a_):
```
[armeabi-v7a] Compile++ thumb: hello-jni <= test.cpp
<snip> \
-march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=softfp -fno-exceptions -fno-rtti -mthumb \
-Os -g -DNDEBUG -fomit-frame-pointer -fno-strict-aliasing -finline-limit=64 \
-O0 -UNDEBUG -marm -fno-omit-frame-pointer \
<snip> \
-c  jni/test.cpp -o ./obj/local/armeabi-v7a/objs-debug/hello-jni/test.o 
```
`-marm` option goes after `-mthumb`.